caom2utils/caom2utils/fits2caom2.py

```diff
@@ -2574,28 +2588,32 @@ def _get_algorithm(self, obs):
         # TODO DEFAULT VALUE
         name = self._get_from_list('Observation.algorithm.name', index=0,
                                    current=obs.algorithm.name)
-        self.logger.debug('End Algorithm augmentation.')
+        result = None
         if name:
-            return Algorithm(str(name))
-        else:
-            return None
+            result = Algorithm(str(name))
+        self.logger.debug('End Algorithm augmentation.')
+        return result
 
-    def _get_instrument(self):
+    def _get_instrument(self, current):
         """
         Create an Instrument instance populated with available FITS
         information.
         :return: Instrument
         """
         self.logger.debug('Begin Instrument augmentation.')
-        name = self._get_from_list('Observation.instrument.name', index=0)
-        keywords = self._get_from_list('Observation.instrument.keywords',
-                                       index=0)
+        if current is None:
```

Comment:
Suggestion for a slightly less verbose form:
`name = self._get_from_list('Observation.instrument.name', index=0, current=None if current is None else current.name)`

Reply:
Changing everywhere as we discussed on slack.